### ISSUE TITLE
Crash on macOS Catalyst while collapsing any InlineRow

### ISSUE THREAD

**Issue by jurex** [xmartlabs/Eureka#2023]
Eureka v5.2.1
Xcode 11.4.1
macOS Catalina 10.15.4
MacBook Pro 13-inch, 2018

Out of bounds exceptions when trying to collapse any inline row (i.e. DateInlineRow, ...)
Thread 1: Fatal error: Section: Index out of bounds

/Pods/Eureka/Source/Core/Section.swift, line 247
<img width="941" alt="Screenshot 2020-04-22 at 10 09 49" src="https://user-images.githubusercontent.com/925257/79957362-bde88000-8481-11ea-99e0-4d0c6fc5d531.png">

**Comment by phr85** [comment]
+1

**Comment by phr85** [comment]
@jurex Could you solve this issue or at least some kind of workaround?

**Comment by jurex** [comment]
nope, the only workaround I found for now is to remove all inline rows.

**Comment by Schlaubischlump** [comment]
Same problem on macOS Catalina 10.15.3

**Comment by gossmer** [comment]
I found this works around it for now. This doesn't seem like the right solution but it works around this problem in a brute force type of way. Edit Core.swift:

   open func rowsHaveBeenAdded(_ rows: [BaseRow], at indexes: [IndexPath]) {
        guard animateTableView else { return }
        
        // Picker InlineRow rows will crash on close with Eureka when running on the mac.
#if targetEnvironment(macCatalyst)
        for row in rows {
           if row is BaseInlineRowType || row is _DatePickerRow || row.baseCell is UIPickerViewDataSource {
                tableView?.reloadData()
                return
            }
        }
#endif

        tableView?.beginUpdates()
        tableView?.insertRows(at: indexes, with: insertAnimation(forRows: rows))
        tableView?.endUpdates()
    }

    open func rowsHaveBeenRemoved(_ rows: [BaseRow], at indexes: [IndexPath]) {
        guard animateTableView else { return }
        
        // Picker InlineRow rows will crash on close with Eureka when running on the mac.
#if targetEnvironment(macCatalyst)
        for row in rows {
           if row is BaseInlineRowType || row is _DatePickerRow || row.baseCell is UIPickerViewDataSource {
                tableView?.reloadData()
                return
            }
        }
#endif
        
        tableView?.beginUpdates()
        tableView?.deleteRows(at: indexes, with: deleteAnimation(forRows: rows))
        tableView?.endUpdates()
    }

**Comment by miajed** [comment]
> +1

+1

**Comment by mats-claassen** [comment]
Does the workaround in #2103 work for you?
